61103 add default autoload core

pbearne opened the pull request:
Trac ticket: https://core.trac.wordpress.org/ticket/61103

Commit:
Change 'autoload' option values in various files

Updated values of 'autoload' option from 'yes' or 'no' to 'on' or 'off' respectively, across several PHP files. This change was implemented in the functions handling options and lock settings, enhancing clarity and consistency of the code.

Commit:
Update script compression settings in ajax-actions

The commit modifies the 'can_compress_scripts' options in the ajax-actions.php file. The update_site_option and update_option functions now have a third argument which controls whether the setting is on or off. This enhances flexibility and control over the script compression feature.

Commit:
Update autoload parameters in option functions

The commit adjusts the parameters for 'update_option' functions in 'featured-content.php' and 'ajax-actions.php'. It changes the 'autoload' parameter where necessary, updating the values for "featured-content" and "can_compress_scripts" options. It's done to ensure the correct use of the autoloading feature in WordPress functionality.

Commit:
Adjust third argument in update_option() across various scripts

This commit adds a third argument to the update_option() function calls across a large number of scripts. This argument toggles whether the option should be loaded from cache (on) or not (off). The changes made help to better manage data caching in WordPress and potentially improve performance by avoiding unnecessary database queries.

Commit:
Adjust third argument in update_option() across various scripts

This commit adds a third argument to the update_option() function calls across a large number of scripts. This argument toggles whether the option should be loaded from cache (on) or not (off). The changes made help to better manage data caching in WordPress and potentially improve performance by avoiding unnecessary database queries.

Commit:
revert of block autoload settings to upgrade.php

mukeshpanchal27 commented on src/wp-admin/includes/upgrade.php:
```diff
@@ -1788,7 +1788,7 @@ function upgrade_330() {
 			unset( $_sidebars_widgets );
 
-			// Intentional fall-through to upgrade to the next version.
+		// Intentional fall-through to upgrade to the next version.
```
```suggestion
			// Intentional fall-through to upgrade to the next version.
```
PHPCS fix

adamsilverstein commented on src/wp-admin/includes/ajax-actions.php:
```diff
@@ -231,15 +231,15 @@ function wp_ajax_wp_compression_test() {
 				update_site_option( 'can_compress_scripts', 0 );
 			} else {
-				update_option( 'can_compress_scripts', 0, 'yes' );
+				update_option( 'can_compress_scripts', 0, 'on' );
```
update_option accepts 'yes' or 'no' or true/false. Why does this PR use 'on' and 'off'?

swissspidy replied:
`on` and `off` are new additions in 6.6. See https://core.trac.wordpress.org/ticket/42441 / https://core.trac.wordpress.org/changeset/57920

joemcgill replied:
That's not exactly correct. `"on"` and `"off"` are the stored DB values, but the function should use boolean `true`/`false` values. For backwards compat, we still support `"yes"` and `"no"`.

pbearne committed:
Update src/wp-admin/includes/upgrade.php

Co-authored-by: Mukesh Panchal <mukeshpanchal27@users.noreply.github.com>

pbearne committed:
Merge branch 'WordPress:trunk' into 61103-add-default-autoload-core

Commit:
Replace 'off'/'on' strings with boolean values in update_option calls

Updated several calls to the update_option function throughout the codebase to use boolean values for autoload parameter instead of 'off'/'on' strings. This change improves the consistency and readability of the related database entries.

pbearne commented:
I have replaced the on/off with booleans

As I have gone through these I have set any option that looks like it is wp-admin only to not autoload this should help the front-end load a little bit

Commit:
Replace 'off'/'on' strings with boolean values in update_option calls

Updated several calls to the update_option function throughout the codebase to use boolean values for autoload parameter instead of 'off'/'on' strings. This change improves the consistency and readability of the related database entries.

Commit:
Update WordPress version and database upgrade procedure

Updated the WordPress database version in `version.php` and performed changes in `upgrade.php` to reflect this. Introduced a new function `upgrade_670()` to execute changes made in WordPress version 6.7.0. Additionally, adjusted the indentation in numerous lines for code readability and consistency.

Commit:
Add transient validation methods and corresponding tests

A new function named 'valid_transient' has been introduced to verify if a transient and its value are valid, also by checking if they have not expired. Sites with transient data can use the 'valid_site_transient' function to perform similar checks. Along with these functions, PHPUnit test cases have been added to ensure these functions perform as expected.

Commit:
Revert "Add transient validation methods and corresponding tests"

This reverts commit 014cf5d7322c4a9ccf58d399f3cb8fe39d855732.

Commit:
Merge branch 'refs/heads/trunk' into 61103-add-default-autoload-core

pbearne committed:
Merge branch 'WordPress:trunk' into 61103-add-default-autoload-core

Commit:
Add caching for network options in wp-admin

A new function, wp_prime_wp_admin_option_caches, has been created to prime specific network options for the current network into the cache using a single database query. Additionally, the action 'admin_init' has been added to autoload these admin options. This should lead to improved performance by reducing the number of database queries.

Commit:
Merge remote-tracking branch 'origin/61103-add-default-autoload-core' into 61103-add-default-autoload-core

pbearne committed:
Merge branch 'WordPress:trunk' into 61103-add-default-autoload-core

pbearne committed:
Update class-wp-widget.php

mukeshpanchal27 commented on src/wp-admin/includes/upgrade.php:
```diff
@@ -988,7 +991,7 @@ function upgrade_110() {
 	$time_difference = $all_options->time_difference;
 
-		$server_time = time() + gmdate( 'Z' );
+	$server_time 	 = time() + gmdate( 'Z' );
 	$weblogger_time  = $server_time + $time_difference * HOUR_IN_SECONDS;
```
```suggestion
	$server_time     = time() + gmdate( 'Z' );
	$weblogger_time  = $server_time + $time_difference * HOUR_IN_SECONDS;
```

Fix PHPCS

mukeshpanchal27 commented on src/wp-admin/includes/upgrade.php:
```diff
@@ -2396,7 +2399,45 @@ function upgrade_650() {
+			'admin_email_lifespan',
+		);
 
+		$autoload = array_fill_keys( $options_ids, 'no' );
+		wp_set_option_autoload_values( $autoload );
```
```suggestion
		wp_set_options_autoload( $options_ids, 'no' );
```
The function will take care.

mukeshpanchal27 commented on src/wp-admin/includes/upgrade.php:
```diff
@@ -2396,7 +2399,45 @@ function upgrade_650() {
 	}
 }
+/**
+ * Executes changes made in WordPress 6.5.0.
```
```suggestion
 * Executes changes made in WordPress 6.7.0.
```

pbearne committed:
Update src/wp-admin/includes/upgrade.php

Co-authored-by: Mukesh Panchal <mukeshpanchal27@users.noreply.github.com>

pbearne committed:
Update src/wp-admin/includes/upgrade.php

Co-authored-by: Mukesh Panchal <mukeshpanchal27@users.noreply.github.com>

pbearne committed:
Update src/wp-admin/includes/upgrade.php

Co-authored-by: Mukesh Panchal <mukeshpanchal27@users.noreply.github.com>

mukeshpanchal27 commented on src/wp-admin/includes/ajax-actions.php:
```diff
@@ -2487,7 +2487,7 @@ function wp_ajax_delete_inactive_widgets() {
 		$widget       = get_option( 'widget_' . $id_base );
 		unset( $widget[ $multi_number ] );
-		update_option( 'widget_' . $id_base, $widget );
+		update_option( 'widget_' . $id_base, $widget, false );
```
Nice!

mukeshpanchal27 commented on src/wp-admin/includes/class-plugin-upgrader.php:
```diff
@@ -262,7 +262,7 @@ public function upgrade( $plugin, $args = array() ) {
 		if ( isset( $past_failure_emails[ $plugin ] ) ) {
 			unset( $past_failure_emails[ $plugin ] );
-			update_option( 'auto_plugin_theme_update_emails', $past_failure_emails );
+			update_option( 'auto_plugin_theme_update_emails', $past_failure_emails, false );
```
I don't think we needs to change it, In https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-admin/includes/schema.php#L578C3-L613 core will set `off`.

mukeshpanchal27 commented on src/wp-includes/option.php:
```diff
@@ -659,6 +659,57 @@ function wp_prime_site_option_caches( array $options ) {
+ *
+ * If site is not multisite, then call wp_prime_option_caches().
+ *
+ * @since 6.6.0
```
```suggestion
 * @since 6.7.0
```

mukeshpanchal27 commented on src/wp-includes/option.php:
```diff
@@ -659,6 +659,57 @@ function wp_prime_site_option_caches( array $options ) {
+	 *
+	 * @since 6.7.0
+	 *
+	 * @param array $default_core_admin_options Array with admin options to be loaded.
+	 *
+	 * @return array Array with admin options to be loaded.
```
```suggestion
	 * @param array $default_core_admin_options Array with admin options to be loaded.
	 * @return array Array with admin options to be loaded.
```

mukeshpanchal27 reviewed (commented):
Thanks @pbearne for the updates. Left some feedbacks.

paul commented:
Could you fix your bot that keeps mentioning me? I'm getting dozens of notifications a month from this project. https://github.com/WordPress/wordpress-develop/pulls?q=sort%3Aupdated-desc+%22%40paul%22

pbearne committed:
Update src/wp-includes/option.php

Co-authored-by: Mukesh Panchal <mukeshpanchal27@users.noreply.github.com>

pbearne committed:
Update src/wp-includes/option.php

Co-authored-by: Mukesh Panchal <mukeshpanchal27@users.noreply.github.com>

peterwilsoncc commented:
> Could you fix your bot that keeps mentioning me? I'm getting dozens of notifications a month from this project. https://github.com/WordPress/wordpress-develop/pulls?q=sort%3Aupdated-desc+%22%40paul%22

@pbearne Are you able to look at the details in your git config so the bot stops pinging the wrong Paul. It's currently set to `paul bearne <paul>` which is causing the bot to be confused.

If you add `pbearne@git.wordpress.org` to your GitHub profile and set that as the email address for WordPress-Develop the commits will be assigned correctly.

pbearne commented:
> If you add `pbearne@git.wordpress.org` to your GitHub profile and set that as the email address for WordPress-Develop the commits will be assigned correctly.

updated

felixarntz commented on src/wp-admin/includes/class-plugin-upgrader.php:
```diff
@@ -262,7 +262,7 @@ public function upgrade( $plugin, $args = array() ) {
 		if ( isset( $past_failure_emails[ $plugin ] ) ) {
 			unset( $past_failure_emails[ $plugin ] );
-			update_option( 'auto_plugin_theme_update_emails', $past_failure_emails );
+			update_option( 'auto_plugin_theme_update_emails', $past_failure_emails, false );
```
The `auto_plugin_theme_update_emails` option is set during WordPress installation (with autoload `false`), so there's no need to specify `false` here. The default will ensure it remains the autoload value it was previously using.

Please change that throughout the rest of the PR as well.

pbearne replied:
done

felixarntz commented on src/wp-admin/includes/misc.php:
```diff
@@ -352,7 +352,7 @@ function update_recently_edited( $file ) {
 	}
 
-	update_option( 'recently_edited', $oldfiles );
+	update_option( 'recently_edited', $oldfiles, false );
```
See above: This is already set to `false` during installation, so no need to pass this here.

pbearne replied:
fixed

felixarntz commented on src/wp-admin/includes/plugin.php:
```diff
@@ -711,7 +711,7 @@ function activate_plugin( $plugin, $redirect = '', $network_wide = false, $silen
 			$current[] = $plugin;
 			sort( $current );
-			update_option( 'active_plugins', $current );
+			update_option( 'active_plugins', $current, false );
```
This looks problematic to me. The active plugins are fetched on basically every single WordPress request, so this _needs_ to be autoloaded.

However, similar to my previous comments, the value here should be omitted entirely, since this option is already populated during WordPress installation (with autoload `true`), so it'll always be correct.

pbearne replied:
fixed

felixarntz commented on src/wp-admin/includes/plugin.php:
```diff
@@ -1304,7 +1304,7 @@ function uninstall_plugin( $plugin ) {
 		if ( isset( $uninstallable_plugins[ $file ] ) ) {
 			unset( $uninstallable_plugins[ $file ] );
-			update_option( 'uninstall_plugins', $uninstallable_plugins );
+			update_option( 'uninstall_plugins', $uninstallable_plugins, false );
```
See above, this is already handled by the WordPress installation, so it should be omitted.

pbearne replied:
fixed

felixarntz commented on src/wp-admin/includes/schema.php:
```diff
@@ -614,7 +614,7 @@ function populate_options( array $options = array() ) {
 	// In case it is set, but blank, update "home".
 	if ( ! __get_option( 'home' ) ) {
-		update_option( 'home', $guessurl );
+		update_option( 'home', $guessurl, true );
```
See above, this should be removed as it is configured during WordPress installation.

pbearne replied:
fixed

felixarntz commented on src/wp-admin/includes/upgrade.php:
```diff
@@ -2396,7 +2399,44 @@ function upgrade_650() {
+
+	if ( $wp_current_db_version < 57160 ) {
+		$options_ids = array(
+			'auto_plugin_theme_update_emails',
```
This is already set to not autoload during installation, so it should be removed from here.

felixarntz commented on src/wp-admin/includes/upgrade.php:
```diff
@@ -2396,7 +2399,44 @@ function upgrade_650() {
+			'recently_edited',
+			'adminhash',
+			'nav_menu_options',
+			'active_plugins',
```
See above, this must be autoloaded, so needs to be removed from here.

felixarntz commented on src/wp-admin/includes/upgrade.php:
```diff
@@ -2396,7 +2399,44 @@ function upgrade_650() {
+			'adminhash',
+			'nav_menu_options',
+			'active_plugins',
+			'uninstall_plugins',
```
See above, this is already not autoloaded during installation, so should be removed.

felixarntz commented on src/wp-admin/includes/upgrade.php:
```diff
@@ -2396,7 +2399,44 @@ function upgrade_650() {
+			'admin_email_lifespan',
+		);
 
+		wp_set_options_autoload( $options_ids, 'no' );
```
Let's use the boolean, as that is encouraged now. Related: https://core.trac.wordpress.org/ticket/61929

felixarntz commented on src/wp-admin/network/site-info.php:
```diff
@@ -98,7 +98,7 @@
 	if ( $old_home_host === $existing_details->domain && $old_home_parsed['path'] === $existing_details->path ) {
 		$new_home_url = untrailingslashit( sanitize_url( $blog_data['scheme'] . '://' . $new_details->domain . $new_details->path ) );
-		update_option( 'home', $new_home_url );
+		update_option( 'home', $new_home_url, true );
```
See above, let's remove.

felixarntz commented on src/wp-admin/network/site-info.php:
```diff
@@ -107,7 +107,7 @@
 	if ( $old_site_host === $existing_details->domain && $old_site_parsed['path'] === $existing_details->path ) {
 		$new_site_url = untrailingslashit( sanitize_url( $blog_data['scheme'] . '://' . $new_details->domain . $new_details->path ) );
-		update_option( 'siteurl', $new_site_url );
+		update_option( 'siteurl', $new_site_url, true );
```
See above, let's remove.

felixarntz commented on src/wp-admin/network/site-settings.php:
```diff
@@ -46,7 +46,7 @@
 			continue; // Avoids "0 is a protected WP option and may not be modified" error when editing blog options.
 		}
-		update_option( $key, $val );
+		update_option( $key, $val, true );
```
This should remain without a value because we have no idea what the option is and therefore whether or not to autoload it. Alternatively, explicitly pass `null` to indicate the value should be left as is.

felixarntz commented on src/wp-admin/options-privacy.php:
```diff
@@ -52,7 +52,7 @@ static function ( $body_class ) {
 	if ( 'set-privacy-page' === $action ) {
 		$privacy_policy_page_id = isset( $_POST['page_for_privacy_policy'] ) ? (int) $_POST['page_for_privacy_policy'] : 0;
-		update_option( 'wp_page_for_privacy_policy', $privacy_policy_page_id );
+		update_option( 'wp_page_for_privacy_policy', $privacy_policy_page_id, true );
```
See above, this is already set during installation and therefore should be removed.

felixarntz commented on src/wp-admin/options-reading.php:
```diff
@@ -74,12 +74,12 @@
 	<?php
 	if ( 'posts' !== get_option( 'show_on_front' ) ) :
-		update_option( 'show_on_front', 'posts' );
+		update_option( 'show_on_front', 'posts', true );
```
See above, this should be removed as it is set during installation.

felixarntz commented on src/wp-admin/options.php:
```diff
@@ -63,7 +63,7 @@
 		&& ! empty( $new_admin_details['newemail'] )
 	) {
-		update_option( 'admin_email', $new_admin_details['newemail'] );
+		update_option( 'admin_email', $new_admin_details['newemail'], false );
```
Not sure if this is safe to _not_ autoload. It is set during WordPress installation to be autoloaded, so we should definitely not override this here without further consideration.

I think it's best to revert this and stick to the current default passed during installation.

felixarntz commented on src/wp-admin/options.php:
```diff
@@ -341,7 +341,7 @@
 				$value = wp_unslash( $value );
 			}
-			update_option( $option, $value );
+			update_option( $option, $value, true );
```
This should definitely be left out. We don't know what the option is so we don't know whether to autoload it. Either this should be removed or we explicitly pass `null` to leave the value as is, or leave the decision to Core.

felixarntz commented on src/wp-content/themes/twentyfourteen/inc/featured-content.php:
```diff
@@ -268,7 +268,7 @@ public static function delete_post_tag( $tag_id ) {
 		$settings['tag-id'] = 0;
 		$settings           = self::validate_settings( $settings );
-		update_option( 'featured-content', $settings );
+		update_option( 'featured-content', $settings, true );
```
This is a default theme change, let's leave those out of the PR as it should only be concerned with update _core's_ usages.

felixarntz commented on src/wp-includes/class-wp-customize-setting.php:
```diff
@@ -656,7 +656,7 @@ protected function set_root_value( $value ) {
 				$autoload = self::$aggregated_multidimensionals[ $this->type ][ $this->id_data['base'] ]['autoload'];
 			}
-			return update_option( $id_base, $value, $autoload );
+			return update_option( $id_base, $value, $autoload, true );
```
This is invalid, the `$autoload` is already the 3rd parameter. Please revert.

felixarntz commented on src/wp-includes/class-wp-xmlrpc-server.php:
```diff
@@ -4330,7 +4330,7 @@ public function wp_setOptions( $args ) {
 			}
 
-			update_option( $this->blog_options[ $o_name ]['option'], wp_unslash( $o_value ) );
+			update_option( $this->blog_options[ $o_name ]['option'], wp_unslash( $o_value ), true );
```
See above, we don't know what the options are so we shouldn't make a decision here on whether to autoload them. This should be reverted.

felixarntz commented on src/wp-includes/ms-blogs.php:
```diff
@@ -460,11 +460,11 @@ function update_blog_option( $id, $option, $value, $deprecated = null ) {
 
 	if ( get_current_blog_id() === $id ) {
-		return update_option( $option, $value );
+		return update_option( $option, $value, true );
```
See above, please revert.

felixarntz commented on src/wp-includes/ms-blogs.php:
```diff
@@ -460,11 +460,11 @@ function update_blog_option( $id, $option, $value, $deprecated = null ) {
 
 	switch_to_blog( $id );
-	$return = update_option( $option, $value );
+	$return = update_option( $option, $value, true );
```
See above, please revert.

felixarntz commented on src/wp-includes/ms-deprecated.php:
```diff
@@ -642,17 +642,17 @@ function install_blog( $blog_id, $blog_title = '' ) {
-	update_option( 'siteurl', $siteurl );
-	update_option( 'home', $home );
+	update_option( 'siteurl', $siteurl, true );
+	update_option( 'home', $home, true );
```
See above, please revert.

felixarntz commented on src/wp-includes/option.php:
```diff
@@ -659,6 +659,56 @@ function wp_prime_site_option_caches( array $options ) {
+	 */
+	$options = apply_filters( 'wp_prime_wp_admin_option_caches', $default_core_admin_options );
+	wp_prime_network_option_caches( null, $options );
+}
```
I don't think we should implement a function like this. Almost all of these options are used in one or two specific places, so it's wasteful to load all of them throughout all WP Admin. It would effectively replicate the current problem just scoped to WP Admin.

Options should only be loaded where they're needed if possible. I think the few changes of not autoload certain options in this PR already work well without this, as all modified options (except some, where I left comments) are used only in one admin screen or section.

felixarntz commented on src/wp-includes/plugin.php:
```diff
@@ -936,7 +936,7 @@ function register_uninstall_hook( $file, $callback ) {
 	if ( ! isset( $uninstallable_plugins[ $plugin_basename ] ) || $uninstallable_plugins[ $plugin_basename ] !== $callback ) {
 		$uninstallable_plugins[ $plugin_basename ] = $callback;
-		update_option( 'uninstall_plugins', $uninstallable_plugins );
+		update_option( 'uninstall_plugins', $uninstallable_plugins, false );
```
See above, this is already configured during installation.

felixarntz commented on src/wp-includes/version.php:
```diff
@@ -23,7 +23,7 @@
  * @global int $wp_db_version
  */
-$wp_db_version = 57155;
+$wp_db_version = 57160;
```
Why is this change needed?

pbearne replied:
We were updating the DB at one point, reverted

felixarntz commented on src/wp-login.php:
```diff
@@ -515,7 +515,7 @@ function wp_login_viewport_meta() {
 
 	if ( get_option( 'siteurl' ) !== $url ) {
-		update_option( 'siteurl', $url );
+		update_option( 'siteurl', $url, true );
```
See above, this is not needed.

pbearne replied:
reverted

felixarntz commented on src/wp-login.php:
```diff
@@ -619,7 +619,7 @@ function wp_login_viewport_meta() {
 
 			if ( $remind_interval > 0 ) {
-				update_option( 'admin_email_lifespan', time() + $remind_interval );
+				update_option( 'admin_email_lifespan', time() + $remind_interval, false );
```
Like my comment with `admin_email`, let's not change this here as it's a bit risky to anticipate where it is used. Plus it shouldn't differ from how it is set during WordPress installation.

pbearne replied:
fixed

felixarntz commented on src/wp-admin/includes/ajax-actions.php:
```diff
@@ -2487,7 +2487,7 @@ function wp_ajax_delete_inactive_widgets() {
 		$widget       = get_option( 'widget_' . $id_base );
 		unset( $widget[ $multi_number ] );
-		update_option( 'widget_' . $id_base, $widget );
+		update_option( 'widget_' . $id_base, $widget, false );
```
Let's leave this change for https://core.trac.wordpress.org/ticket/42743. It is its own problem to resolve and would make this PR more complicated.

pbearne replied:
I agree

felixarntz commented on src/wp-admin/widgets-form.php:
```diff
@@ -211,7 +211,7 @@
 			$widget       = get_option( 'widget_' . $id_base );
 			unset( $widget[ $multi_number ] );
-			update_option( 'widget_' . $id_base, $widget );
+			update_option( 'widget_' . $id_base, $widget, false );
```
See above, let's leave this for https://core.trac.wordpress.org/ticket/42743.

pbearne replied:
yes

felixarntz reviewed (changes requested):
@pbearne I did a thorough review of all the options you're (potentially) modifying here.

My most important feedback is: We should only add `true` or `false` to `update_option()` for options that aren't already added to the database during WordPress installation. Any such options will always already be set when calling `update_option()`.

It is only mandatory to pass an `$autoload` value to `add_option()` because here we're inserting a new option. On `update_option()`, `$autoload` however is only needed if it _may_ pass through to `add_option()` or if we explicitly want to change the autoload value from what it was before. Otherwise it should be `null`.

Also: We shouldn't pass `true` on `update_option()` calls that are for any arbitrary options. For that, the default `null` is intended, which is to leave it up to WordPress core to make the decision.

I think most of the options that are _not_ part of the WordPress installation and that you changed to no longer autoload, it makes sense because they're only used in 1-2 specific admin sections. There are a few though I left comments on too where I think this is too risky and should be separately considered.

Commit:
Remove deprecated third parameter from update_option calls

Removed the deprecated third parameter from multiple update_option calls across various files for consistency and to avoid warnings. This change ensures that all update_option calls follow the current best practices in WordPress development.

Commit:
Update alloptions_count value in wpSiteHealth test

Changed the value of 'alloptions_count' from 100 to 18 to reflect the accurate count after the expended 20. This ensures the site health test results are precise and consistent with the current state.

Commit:
Update wpGetArchives test to match expected query count

Changed assertions in the wpGetArchives test to reflect the correct number of queries being executed. This ensures proper validation of queries after recent changes in the query logic.

pbearne committed:
Merge branch 'WordPress:trunk' into 61103-add-default-autoload-core

Commit:
reverted wrong changes

pbearne committed:
Merge branch 'WordPress:trunk' into 61103-add-default-autoload-core

Commit:
Remove 'autoload' parameter from update_option calls

Refactored various instances of update_option function calls to exclude the 'autoload' parameter for improved code simplicity and performance. Also removed the now redundant wp_prime_wp_admin_option_caches function and its related filter.

Commit:
Merge remote-tracking branch 'origin/61103-add-default-autoload-core' into 61103-add-default-autoload-core

felixarntz commented on src/wp-admin/includes/class-plugin-upgrader.php:
```diff
@@ -444,7 +444,7 @@ public function bulk_upgrade( $plugins, $args = array() ) {
 		}
 
-		update_option( 'auto_plugin_theme_update_emails', $past_failure_emails );
+		update_option( 'auto_plugin_theme_update_emails', $past_failure_emails, false );
```
This still needs to be reverted, see https://github.com/WordPress/wordpress-develop/pull/6465#discussion_r1731865969.
```suggestion
		update_option( 'auto_plugin_theme_update_emails', $past_failure_emails );
```

felixarntz commented on src/wp-admin/includes/misc.php:
```diff
@@ -352,7 +352,7 @@ function update_recently_edited( $file ) {
 	}
 
-	update_option( 'recently_edited', $oldfiles );
+	update_option( 'recently_edited', $oldfiles, false );
```
This still needs to be reverted, see https://github.com/WordPress/wordpress-develop/pull/6465#discussion_r1731876165.
```suggestion
	update_option( 'recently_edited', $oldfiles );
```

felixarntz commented on src/wp-admin/includes/plugin.php:
```diff
@@ -842,7 +842,7 @@ function deactivate_plugins( $plugins, $silent = false, $network_wide = null ) {
 
 	if ( $do_blog ) {
-		update_option( 'active_plugins', $current );
+		update_option( 'active_plugins', $current, false );
```
Per https://github.com/WordPress/wordpress-develop/pull/6465#discussion_r1731880434, this needs to be reverted too.
```suggestion
		update_option( 'active_plugins', $current );
```

felixarntz commented on src/wp-admin/includes/plugin.php:
```diff
@@ -1070,7 +1070,7 @@ function validate_active_plugins() {
 	// Validate vartype: array.
 	if ( ! is_array( $plugins ) ) {
-		update_option( 'active_plugins', array() );
+		update_option( 'active_plugins', array(), false );
```
See above:
```suggestion
		update_option( 'active_plugins', array() );
```

felixarntz commented on src/wp-admin/includes/upgrade.php:
```diff
@@ -2397,7 +2400,41 @@ function upgrade_650() {
+			'_wp_suggested_policy_text_has_changed',
+			'dashboard_widget_options',
+			'ftp_credentials',
+			'recently_edited',
```
Per the previous feedback, this option should not be modified.
```suggestion
```

felixarntz commented on src/wp-admin/includes/upgrade.php:
```diff
@@ -2397,7 +2400,41 @@ function upgrade_650() {
+			'wp_force_deactivated_plugins',
+			'delete_blog_hash',
+			'allowedthemes',
+			'admin_email',
```
See previous feedback, this still needs to be autoloaded.
```suggestion
```

felixarntz commented on src/wp-admin/includes/upgrade.php:
```diff
@@ -2397,7 +2400,41 @@ function upgrade_650() {
+			'delete_blog_hash',
+			'allowedthemes',
+			'admin_email',
+			'recently_activated',
```
Duplicate, you already have this on top of the list.
```suggestion
```

felixarntz commented on src/wp-admin/includes/upgrade.php:
```diff
@@ -2397,7 +2400,41 @@ function upgrade_650() {
+			'recently_activated',
+			'https_detection_errors',
+			'fresh_site',
+			'upload_path',
+			'admin_email',
+			'admin_email_lifespan',
```
See previous feedback, these should still be autoloaded. On the other hand, the list is missing the `recovery_keys` option.
```suggestion
			'recovery_keys',
```

felixarntz commented on src/wp-admin/options-privacy.php:
```diff
@@ -103,7 +103,7 @@ static function ( $body_class ) {
 			);
 		} else {
-			update_option( 'wp_page_for_privacy_policy', $privacy_policy_page_id );
+			update_option( 'wp_page_for_privacy_policy', $privacy_policy_page_id, true );
```
This still needs to be reverted per https://github.com/WordPress/wordpress-develop/pull/6465#discussion_r1731897669.
```suggestion
			update_option( 'wp_page_for_privacy_policy', $privacy_policy_page_id );
```

felixarntz commented on src/wp-admin/options-reading.php:
```diff
@@ -79,7 +79,7 @@
 else :
 	if ( 'page' === get_option( 'show_on_front' ) && ! get_option( 'page_on_front' ) && ! get_option( 'page_for_posts' ) ) {
-		update_option( 'show_on_front', 'posts' );
+		update_option( 'show_on_front', 'posts', true );
```
https://github.com/WordPress/wordpress-develop/pull/6465#discussion_r1731897851
```suggestion
		update_option( 'show_on_front', 'posts' );
```

felixarntz commented on src/wp-admin/widgets-form.php:
```diff
@@ -211,7 +211,7 @@
 			$widget       = get_option( 'widget_' . $id_base );
 			unset( $widget[ $multi_number ] );
-			update_option( 'widget_' . $id_base, $widget );
+			update_option( 'widget_' . $id_base, $widget, false );
```
This still needs to be addressed per https://github.com/WordPress/wordpress-develop/pull/6465#discussion_r1731918565.
```suggestion
			update_option( 'widget_' . $id_base, $widget );
```

felixarntz commented on src/wp-includes/class-wp-widget.php:
```diff
@@ -598,7 +598,7 @@ public function _register_one( $number = -1 ) {
 	public function save_settings( $settings ) {
 		$settings['_multiwidget'] = 1;
-		update_option( $this->option_name, $settings );
+		update_option( $this->option_name, $settings, wp_use_widgets_block_editor() );
```
Per https://github.com/WordPress/wordpress-develop/pull/6465#discussion_r1731918565, please revert as this should be handled via https://core.trac.wordpress.org/ticket/42743.
```suggestion
		update_option( $this->option_name, $settings );
```

felixarntz commented on src/wp-includes/ms-deprecated.php:
```diff
@@ -646,13 +646,13 @@ function install_blog( $blog_id, $blog_title = '' ) {
 
 	if ( get_site_option( 'ms_files_rewriting' ) ) {
-		update_option( 'upload_path', UPLOADBLOGSDIR . "/$blog_id/files" );
+		update_option( 'upload_path', UPLOADBLOGSDIR . "/$blog_id/files", false );
```
This should remain autoloaded as it may be regularly accessed in multisite.
```suggestion
		update_option( 'upload_path', UPLOADBLOGSDIR . "/$blog_id/files" );
```

felixarntz commented on src/wp-includes/ms-deprecated.php:
```diff
@@ -646,13 +646,13 @@ function install_blog( $blog_id, $blog_title = '' ) {
+		update_option( 'upload_path', UPLOADBLOGSDIR . "/$blog_id/files", false );
 	} else {
-		update_option( 'upload_path', get_blog_option( get_network()->site_id, 'upload_path' ) );
+		update_option( 'upload_path', get_blog_option( get_network()->site_id, 'upload_path' ), false );
```
See above.
```suggestion
		update_option( 'upload_path', get_blog_option( get_network()->site_id, 'upload_path' ) );
```

felixarntz commented on src/wp-includes/ms-deprecated.php:
```diff
@@ -646,13 +646,13 @@ function install_blog( $blog_id, $blog_title = '' ) {
 
-	update_option( 'blogname', wp_unslash( $blog_title ) );
-	update_option( 'admin_email', '' );
+	update_option( 'blogname', wp_unslash( $blog_title ), true );
+	update_option( 'admin_email', '', false );
```
Per previous feedback, these should remain autoloaded as is (and are already set up during installation).
```suggestion
	update_option( 'blogname', wp_unslash( $blog_title ) );
	update_option( 'admin_email', '' );
```

felixarntz commented on src/wp-includes/widgets.php:
```diff
@@ -1099,7 +1099,7 @@ function wp_set_sidebars_widgets( $sidebars_widgets ) {
 	}
 
-	update_option( 'sidebars_widgets', $sidebars_widgets );
+	update_option( 'sidebars_widgets', $sidebars_widgets, true );
```
Better to leave this as is, and handle it in https://core.trac.wordpress.org/ticket/42743 if needed.
```suggestion
	update_option( 'sidebars_widgets', $sidebars_widgets );
```

felixarntz commented on src/wp-login.php:
```diff
@@ -619,7 +619,7 @@ function wp_login_viewport_meta() {
 
 			if ( $remind_interval > 0 ) {
-				update_option( 'admin_email_lifespan', time() + $remind_interval );
+				update_option( 'admin_email_lifespan', time() + $remind_interval, false );
```
This still needs to be reverted per https://github.com/WordPress/wordpress-develop/pull/6465#discussion_r1731915673.
```suggestion
				update_option( 'admin_email_lifespan', time() + $remind_interval );
```

felixarntz commented on src/wp-admin/includes/upgrade.php:
```diff
@@ -2397,7 +2400,41 @@ function upgrade_650() {
+			'admin_email',
+			'recently_activated',
+			'https_detection_errors',
+			'fresh_site',
```
There is one more instance where `'fresh_site'` updates needs to be modified, in `wp-admin/includes/upgrade.php`.

felixarntz replied:
Updated in https://github.com/WordPress/wordpress-develop/pull/6465/commits/28e9a8779a768d17544157b8d39bd71305e2d950

felixarntz reviewed (commented):
@pbearne There are a few more instances from my previous review that still need to be reverted, but it's getting close.

I can make some of these updates inline.

felixarntz committed:
Revert remaining risky autoload changes.

felixarntz committed:
Revert widget autoload modifications in favor of leaving them for ticket 42743.

Commit:
Merge branch 'trunk' into 61103-add-default-autoload-core

Commit:
Consistently set fresh_site option to not autoload.

Commit:
Set DB version.

felixarntz commented:
Committed in https://core.trac.wordpress.org/changeset/58975
